tests/rptest/services/admin.py

```diff
@@ -226,34 +233,45 @@ def partition_transfer_leadership(self, namespace, topic, partition,
         path = f"partitions/{namespace}/{topic}/{partition}/transfer_leadership?target={target_id}"
         self._request("POST", path)
 
+    def get_partition_leader(self, *, namespace, topic, partition):
+        partition_info = self.get_partitions(topic=topic,
+                                             partition=partition,
+                                             namespace=namespace)
+
+        return partition_info['leader_id']
+
     def transfer_leadership_to(self, *, namespace, topic, partition, target):
         """
         Looks up current ntp leader and transfer leadership to target node, 
-        this operations is NOP when current leader is the same as target. 
+        this operations is NOP when current leader is the same as target.
+        If user pass None for target this function will choose random replica for new leader.
```

Comment:
I think that `redpanda` already selects a random replica when no target is specified, but this needs to be confirmed.

Reply:
yep
Inside it does 
```
auto it = std::max_element(
  _fstats.begin(), _fstats.end(), [](const auto& a, const auto& b) {
      return a.second.last_dirty_log_index
             < b.second.last_dirty_log_index;
  });
```

I wanna avoid choosing current leader for new leader. And changed "random" to "next" in nodes array for topic.